framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2677,15 +2663,10 @@ protected boolean checkConstructorInvocation(
         // Issue a warning if the type at constructor invocation is a subtype of the constructor
         // declaration type.
         // This is equivalent to down-casting.
-        // Only check the primary annotations, the type arguments are checked elsewhere.
-        if (!atypeFactory
-                .getQualifierHierarchy()
-                .isSubtype(returnType.getAnnotations(), invocation.getAnnotations())) {
+        if (!atypeFactory.getQualifierHierarchy().isSubtype(returnType, invocationAnnos)) {
             checker.report(
                     Result.warning(
-                            "cast.unsafe.constructor.invocation",
-                            returnType.toString(true),
-                            invocation.toString(true)),
+                            "cast.unsafe.constructor.invocation", returnType, invocationAnnos),
```

Comment:
I think this message would be easier to read if it had the same arguments as the error message above, i.e. include the constructor that is invoked and order the components consistently.